app/models/user.rb

```diff
@@ -22,4 +26,17 @@ def normalize_queue_item_positions
       item.update_attributes(position: index + 1)
     end
   end
+
+  def followable?(leader)
+    self.not_yet_followed?(leader) && self.is_not_leader?(leader)
+  end
+
+  def not_yet_followed?(leader)
+    !self.leaders.include?(leader)
+  end
+
+  def is_not_leader?(leader)
```

Comment:
I don't think this method is needed. You are abstracting `self != leader`. I don't think that warrants abstraction and the indirection that comes with it.